packages/accounts-controller/src/AccountsController.ts

```diff
@@ -248,12 +292,82 @@ export class AccountsController extends BaseController<
   }
 
   /**
-   * Returns the selected internal account.
+   * Returns the last selected evm account.
    *
    * @returns The selected internal account.
    */
   getSelectedAccount(): InternalAccount {
-    return this.getAccountExpect(this.state.internalAccounts.selectedAccount);
+    const selectedAccount = this.getAccountExpect(
+      this.state.internalAccounts.selectedAccount,
+    );
+    if (isEvmAccountType(selectedAccount.type)) {
+      return selectedAccount;
+    }
+
+    const accounts = this.listAccounts();
+
+    if (!accounts.length) {
+      // !Should never reach this.
+      throw new Error('AccountsController: No EVM accounts');
```

Comment:
Since we do not have the prefix for other error messages, do we want one here too?

Reply:
Yea, we can remove this